src/simple-pg/index.js

```diff
@@ -0,0 +1,24 @@
+import { wrap, unwrap } from '@graffy/common';
+import { dbRead, dbWrite } from './transaction';
+import pg from './pool.js';
+
+export default (opts = {}) =>
+  (store) => {
+    store.on('read', read);
+    store.on('write', write);
+
+    function read(query) {
+      return dbRead(query, opts, store);
+    }
+
+    async function write(change) {
+      change = unwrap(change, store.path);
+      await dbWrite(change, opts);
+      return wrap(change, store.path);
+    }
+  };
+
+export const connect = (config) => pg.connect(config);
+export const setPool = (pool) => pg.setPool(pool);
+export const setPgClient = (client) => pg.setClient(client);
```

Comment:
I am not a fan of stateful modules like this. However given we looking into a different approach, happy for it to stay. I would remove the unused ones though (I think only setPool is being used?)

Reply:
I would suggest replacing this also with accepting `client` or `pool` as an option.

```
import Transaction from './transaction';

const pg = ({ client, pool, ...opts }) => (store) => {
  const myTransaction = new Transaction({ client, pool });

  store.on('read', read);
  store.on('write', write);

  function read(query) {
    return myTransaction.dbRead(query, opts, store);
  }

  async function write(change) {
    change = unwrap(change, store.path);
    await myTransaction.dbWrite(change, opts);
    return wrap(change, store.path);
  }
};

pg.customize = (defaultOptions) => (options) => pg({ ...defaultOptions, ...options });
export default pg;
```

That last `customize` function does all the work of your `setX()` functions without the need for stateful modules (which are, as mentioned elsewhere, evil).

Usage:

```js
import pg from '@graffy/pg';
import { Pool } from 'pg';

const pool = new Pool({ DBHOST: ... });
const myPg = pg.customize({ pool });

store.use('user', myPg());
```

Reply:
All the stateful functions were removed

Reply:
I really like this approach, I have removed pg client out of graffy-pg. We should let developers control pg connection.